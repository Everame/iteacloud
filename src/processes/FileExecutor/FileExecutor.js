import React from "react";
import File from "../File/File";
import Folder from "../Folder/Folder";

export default class FileExecutor{

    init(){
        const file = new File("folder", "c://projects/", '1mb', "Empty Folder");
        const testArr = new Array({"type": file.type, "url": file.url, "size": file.size, "name": file.name}, {"type": file.type, "url": file.url, "size": file.size, "name": file.name});
        localStorage.setItem('storage', JSON.stringify(testArr));
    }

    readDir(){
        return JSON.parse(localStorage.getItem('storage'));
    }

    uploadFile(url){
        const storage = localStorage.getItem('storage');
        storage.push(File());
        return 0;
    }

    createFolder(name){
        const folder = new Folder(name);
        const storage = JSON.parse(localStorage.getItem('storage'));
        storage.push(folder);
        localStorage.setItem('storage', JSON.stringify(storage));
    }
}