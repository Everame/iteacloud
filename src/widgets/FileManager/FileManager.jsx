import React, {useState } from 'react'
import FileView from '../../features/FileView/FileView'
import './style/FileManager.scss'
import FileAction from '../../features/FileAction/FileAction'
import FileExecutor from '../../processes/FileExecutor/FileExecutor';
import ToastWrapper from '../ToastWrapper/ToastWrapper';

export default function FileManager() {
  const [folderIndex, setFolderIndex] = useState(-1);
  const [currentIndex, setCurrentIndex] = useState(-2);
  const [toast, setToast] = useState();
  const [isFolder, setIsFolder] = useState(false);
  const fileExec = new FileExecutor();
  const [storage, setStorage] = useState(fileExec.readDir());
  function openFolder(){
    setFolderIndex(currentIndex);
    setCurrentIndex(-1);
    setStorage(fileExec.readDir(currentIndex));
  }
  function returnToRoot(){
    setFolderIndex(-1);
    setStorage(fileExec.readDir());
  }
  return (
    <section className="fileManagerWrapper">
        <FileAction index={currentIndex} setStorage={setStorage} folder={folderIndex} isFolder={isFolder} setToast={setToast} />
        <FileView storage={storage} setIndex={setCurrentIndex} openFolder={openFolder} returnToRoot={returnToRoot} currentIndex={currentIndex} setIsFolder={setIsFolder} folder={folderIndex} />
        <ToastWrapper toast={toast} setToast={setToast} />
    </section>
  )
}
