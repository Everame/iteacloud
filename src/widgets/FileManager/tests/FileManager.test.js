import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import { MemoryRouter } from 'react-router';
import FileManager from '../FileManager';
import FileExecutor from '../../../processes/FileExecutor/FileExecutor';

describe("FileManager", () => {
    it('FileManager is render', () => {  
        const FileExec = new FileExecutor();
        FileExec.reset();
        const manager = render(
            <MemoryRouter>
              <FileManager />
            </MemoryRouter>);
        expect(manager).toMatchSnapshot();
    });
})