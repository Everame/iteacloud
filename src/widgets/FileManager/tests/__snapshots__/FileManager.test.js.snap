// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FileManager FileManager is render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="fileManagerWrapper"
      >
        <div
          class="actionWrapper"
        >
          <a
            class="actionBtn show"
            data-testid="action"
            id=""
          >
            <div
              class="iconBlock"
            >
              <img
                alt="Button icon"
                src="folder-add.svg"
              />
            </div>
            <div
              class="textBlock"
            >
              <span
                class="title"
              >
                Создать папку
              </span>
            </div>
          </a>
          <a
            class="actionBtn show"
            data-testid="action"
            id=""
          >
            <div
              class="iconBlock"
            >
              <img
                alt="Button icon"
                src="upload.svg"
              />
            </div>
            <div
              class="textBlock"
            >
              <span
                class="title"
              >
                Загрузить файл
              </span>
            </div>
          </a>
          <a
            class="actionBtn "
            data-testid="action"
            id="download"
          >
            <div
              class="iconBlock"
            >
              <img
                alt="Button icon"
                src="download.svg"
              />
            </div>
            <div
              class="textBlock"
            >
              <span
                class="title"
              >
                Скачать файл
              </span>
            </div>
          </a>
          <a
            class="actionBtn "
            data-testid="action"
            id=""
          >
            <div
              class="iconBlock"
            >
              <img
                alt="Button icon"
                src="rename.svg"
              />
            </div>
            <div
              class="textBlock"
            >
              <span
                class="title"
              >
                Переименовать файл
              </span>
            </div>
          </a>
          <a
            class="actionBtn "
            data-testid="action"
            id=""
          >
            <div
              class="iconBlock"
            >
              <img
                alt="Button icon"
                src="delete.svg"
              />
            </div>
            <div
              class="textBlock"
            >
              <span
                class="title"
              >
                Удалить файл
              </span>
            </div>
          </a>
          <div
            class="modalWrapper "
            data-testid="modalAction"
          >
            <div
              class="inputWrapper"
            >
              <div
                class="inputLabel"
              >
                Новое наименование:
              </div>
              <input
                placeholder="Новый файл"
                type="text"
                value=""
              />
            </div>
            <div
              class="confirmBtn "
            >
              <span
                class="btnText"
              >
                Переименовать файл
              </span>
            </div>
            <div
              class="closeBtn"
            >
              <img
                alt="Close Icon"
                src="close.svg"
              />
            </div>
          </div>
          <div
            class="darkLayout "
          />
        </div>
        <div
          class="fileViewWrapper"
        >
          <div
            class="filesBlock"
            data-testid="files"
          />
          <div
            class="btnWrapper "
            data-testid="backBtn"
          >
            <img
              alt="Back button"
              src="angle-left.svg"
            />
          </div>
          <div
            class="sizeWrapper"
          >
            <div
              class="valuesRow"
            >
              <div
                class="value"
              >
                <h4>
                  0.000
                  mb
                </h4>
              </div>
              <div
                class="value"
              >
                <h4>
                  3.75
                  mb
                </h4>
              </div>
            </div>
            <div
              class="progressBar"
            >
              <div
                class="innerRow"
                style="width: 0%;"
              />
            </div>
          </div>
        </div>
        <div
          class="toastWrapper "
        />
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="fileManagerWrapper"
    >
      <div
        class="actionWrapper"
      >
        <a
          class="actionBtn show"
          data-testid="action"
          id=""
        >
          <div
            class="iconBlock"
          >
            <img
              alt="Button icon"
              src="folder-add.svg"
            />
          </div>
          <div
            class="textBlock"
          >
            <span
              class="title"
            >
              Создать папку
            </span>
          </div>
        </a>
        <a
          class="actionBtn show"
          data-testid="action"
          id=""
        >
          <div
            class="iconBlock"
          >
            <img
              alt="Button icon"
              src="upload.svg"
            />
          </div>
          <div
            class="textBlock"
          >
            <span
              class="title"
            >
              Загрузить файл
            </span>
          </div>
        </a>
        <a
          class="actionBtn "
          data-testid="action"
          id="download"
        >
          <div
            class="iconBlock"
          >
            <img
              alt="Button icon"
              src="download.svg"
            />
          </div>
          <div
            class="textBlock"
          >
            <span
              class="title"
            >
              Скачать файл
            </span>
          </div>
        </a>
        <a
          class="actionBtn "
          data-testid="action"
          id=""
        >
          <div
            class="iconBlock"
          >
            <img
              alt="Button icon"
              src="rename.svg"
            />
          </div>
          <div
            class="textBlock"
          >
            <span
              class="title"
            >
              Переименовать файл
            </span>
          </div>
        </a>
        <a
          class="actionBtn "
          data-testid="action"
          id=""
        >
          <div
            class="iconBlock"
          >
            <img
              alt="Button icon"
              src="delete.svg"
            />
          </div>
          <div
            class="textBlock"
          >
            <span
              class="title"
            >
              Удалить файл
            </span>
          </div>
        </a>
        <div
          class="modalWrapper "
          data-testid="modalAction"
        >
          <div
            class="inputWrapper"
          >
            <div
              class="inputLabel"
            >
              Новое наименование:
            </div>
            <input
              placeholder="Новый файл"
              type="text"
              value=""
            />
          </div>
          <div
            class="confirmBtn "
          >
            <span
              class="btnText"
            >
              Переименовать файл
            </span>
          </div>
          <div
            class="closeBtn"
          >
            <img
              alt="Close Icon"
              src="close.svg"
            />
          </div>
        </div>
        <div
          class="darkLayout "
        />
      </div>
      <div
        class="fileViewWrapper"
      >
        <div
          class="filesBlock"
          data-testid="files"
        />
        <div
          class="btnWrapper "
          data-testid="backBtn"
        >
          <img
            alt="Back button"
            src="angle-left.svg"
          />
        </div>
        <div
          class="sizeWrapper"
        >
          <div
            class="valuesRow"
          >
            <div
              class="value"
            >
              <h4>
                0.000
                mb
              </h4>
            </div>
            <div
              class="value"
            >
              <h4>
                3.75
                mb
              </h4>
            </div>
          </div>
          <div
            class="progressBar"
          >
            <div
              class="innerRow"
              style="width: 0%;"
            />
          </div>
        </div>
      </div>
      <div
        class="toastWrapper "
      />
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
